cypress-io/cypress
# chore: upgrade Lerna to v5 and use Nx

- Closes https://github.com/cypress-io/cypress/issues/23688

### Additional details


This PR upgrades Lerna to version 5.4.3. This is the newest version that we can upgrade to without needing to upgrade any of our other dependencies, and still gives us the ability to use Nx as the task runner under the hood of Lerna. It also gives us the ability to view [Nx's dependency graph](https://nx.dev/core-features/explore-graph) for our project to analyze each package's dependencies and the structure of our monorepo. 

This should be a good starting point that will allow us to move forward with leveraging Nx features like [task result caching](https://nx.dev/core-features/cache-task-results), [task distribution](https://nx.dev/core-features/distribute-task-execution) and others to improve our build and test processes.

### Steps to test


Run `yarn` and make sure that everything still works as expected - console output has changed slightly.

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
